satcom/csp/csp_packet.py

```diff
@@ -0,0 +1,188 @@
+from io import BytesIO
+from pydantic import BaseModel
+from satcom.utils import utils
+
+HEADER_LENGTH_BYTES = 4
+# field lengths (# bits)
+FLEN_PRIO  = 2
+FLEN_ADDR  = 5
+FLEN_PORT  = 6
+FLEN_FLAGS = 8
+
+class CSPPacketHeader(BaseModel):
+    # 2 bits, conventionally
+    # 0 (cricical), 1 (high), 2 (norm), 3 (low)
+    priority: int = 0
+
+    # 5 bits: 0 - 31
+    destination: int = 0
+    source: int = 0
+
+    # 6 bits: 0 - 63
+    destination_port: int = 0
+    source_port: int = 0
+
+    #flags: int = 0
+
+    def err(self):
+        """Throws an error if any params are out of bounds"""
+        if self.priority < 0 or self.priority > 3:
+            return ValueError('PacketHeader.priority must be 0-3')
+        if self.destination < 0 or self.destination > 31:
+            return ValueError('PacketHeader.destination must be 0-31')
+        if self.source < 0 or self.source > 31:
+            return ValueError('PacketHeader.source must be 0-31')
+        if self.destination_port < 0 or self.destination_port > 63:
+            return ValueError('PacketHeader.destination_port must be 0-63')
+        if self.source_port < 0 or self.source_port > 63:
+            return ValueError('PacketHeader.source_port must be 0-63')
+        return None
+
+    def to_bytes(self):
+        """Packs CSP packet header metadata into a bytearray"""
+        header = 0
+        cursor = 0
+        bitmask32 = 0xFFFFFFFF
+
+        cursor += FLEN_PRIO
+        header |= (self.priority << (32 - cursor)) & bitmask32
+
+        cursor += FLEN_ADDR
+        header |= (self.source << (32 - cursor)) & bitmask32
+
+        cursor += FLEN_ADDR
+        header |= (self.destination << (32 - cursor)) & bitmask32
+
+        cursor += FLEN_PORT
+        header |= (self.destination_port << (32 - cursor)) & bitmask32
+
+        cursor += FLEN_PORT
+        header |= (self.source_port << (32 - cursor)) & bitmask32
+
+        cursor += FLEN_FLAGS
+        header |= (0 << (32 - cursor)) & bitmask32
+
+        bs = bytearray(HEADER_LENGTH_BYTES)
+        bs = utils.pack_uint_big_endian(header)
+
+        return bs
+
+    @classmethod
+    def from_bytes(cls, bs: bytearray):
```

Comment:
Let's use `bytes` type wherever we accept or return data (see updates to https://github.com/antaris-inc/python-satcom/blob/main/satcom/openlst/space_packet_lib.py, for example). We can use `bytearray` within the functions for read/write operations but I think it will protect everyone if we use `bytes` when interfacing with external clients